src/test/java/org/folio/linked/data/integration/kafka/listener/handler/InventoryInstanceEventHandlerTest.java

```diff
@@ -0,0 +1,129 @@
+package org.folio.linked.data.integration.kafka.listener.handler;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.folio.linked.data.domain.dto.InventoryInstanceEvent;
+import org.folio.linked.data.domain.dto.ResourceIndexEvent;
+import org.folio.linked.data.mapper.kafka.inventory.InventoryInstanceMapper;
+import org.folio.linked.data.repo.FolioMetadataRepository;
+import org.folio.linked.data.test.TestUtil;
+import org.folio.spring.testing.type.UnitTest;
+import org.folio.spring.tools.kafka.FolioMessageProducer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.context.ApplicationEventPublisher;
+
+@UnitTest
+@ExtendWith(MockitoExtension.class)
+public class InventoryInstanceEventHandlerTest {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper()
+    .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+  @Mock
+  private FolioMessageProducer<ResourceIndexEvent> bibliographicMessageProducer;
+  @Mock
+  private ApplicationEventPublisher eventPublisher;
+  @Mock
+  private InventoryInstanceMapper inventoryInstanceMapper;
+  @Mock
+  private FolioMetadataRepository folioMetadataRepository;
+  @InjectMocks
+  private InventoryInstanceEventHandler handler;
+
+  private void verifyNoReindexInteractions() {
+    verifyNoInteractions(inventoryInstanceMapper);
+    verifyNoInteractions(bibliographicMessageProducer);
+    verifyNoInteractions(eventPublisher);
+  }
+
+  @Test
+  void shouldNotReindex_ifSuppressFlagsNotChanged() throws JsonProcessingException {
+    // given
+    var event = getEventWithNotChangesInSuppressFlags();
+
+    // when
+    handler.handle(event);
+
+    // then
+    verifyNoReindexInteractions();
+  }
+
+  @Test
+  void shouldNotReindex_ifInstanceIsNull() throws JsonProcessingException {
+    // given
+    var event = getEventWithNullableObject();
+
+    // when
+    handler.handle(event);
+
+    // then
+    verifyNoReindexInteractions();
+  }
+
+  @Test
+  void shouldNotReindex_ifInstanceNotExistsInRepo() {
+    // given
+    var event = getEvent();
+    when(folioMetadataRepository.existsByInventoryId(any())).thenReturn(false);
+
+    // when
+    handler.handle(event);
+
+    // then
+    verify(folioMetadataRepository).existsByInventoryId(any());
```

Comment:
this line looks redundant

Reply:
it's needed to check that underlying call of existsByInventoryId is made.